spring-pulsar/src/test/java/org/springframework/pulsar/core/PulsarTemplateTests.java

```diff
@@ -17,110 +17,111 @@
 package org.springframework.pulsar.core;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
 
-import java.util.HashMap;
+import java.time.Duration;
+import java.util.Collections;
 import java.util.Map;
 import java.util.concurrent.CompletableFuture;
-import java.util.concurrent.ExecutionException;
 import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
 
+import org.apache.pulsar.client.admin.PulsarAdmin;
 import org.apache.pulsar.client.api.Consumer;
 import org.apache.pulsar.client.api.Message;
 import org.apache.pulsar.client.api.MessageId;
-import org.apache.pulsar.client.api.Producer;
+import org.apache.pulsar.client.api.MessageRouter;
 import org.apache.pulsar.client.api.PulsarClient;
+import org.apache.pulsar.client.api.PulsarClientException;
 import org.apache.pulsar.client.api.Schema;
-import org.junit.jupiter.api.Test;
+import org.apache.pulsar.client.api.TopicMetadata;
+import org.assertj.core.api.InstanceOfAssertFactories;
+import org.junit.jupiter.api.Named;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
 
 /**
+ * Tests for {@code PulsarTemplate}.
+ *
  * @author Soby Chacko
+ * @author Chris Bono
```

Comment:
@soby we can go over this one via Zoom if you like. It basically removes the "testUsage" test and replaces the 2 sync/async cases w/ the permutations of `(syncOrAsync, defaultOrSpecificTopic, withOrWithoutRouter)` via `@ParameterizedTest`.

Reply:
@sobychacko

Reply:
My bad @soby . Sorry for the interruption - have a nice day. 😸